### PR TITLE
Use Amazon's Shopping Queries Dataset

### PR DESCRIPTION
**Pull request by ukohank517**
ref: https://github.com/chlorochrule/solrcloud-development-box/issues/18

## what I do
1. take amazon shopping data as a submodule
2. add a script to generate shopping json data, and create configset files so that json data can be feed to solr system.(but not sure for the field definition)
3. add a script to generate queries
4. add readme, to introduce all above

### CODE REVIEW

**Comment by chlorochrule** [comment]
hint: [a configset](https://github.com/chlorochrule/solrcloud-development-box/tree/main/solr/configsets/wikipedia) for esci dataset is required :)

**Comment by ukohank517** [comment]
here is the amazon data sample,
sorry I couldn't find the relevance between query and other fields
```json
[
  {
    "example_id": 0,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B000MOO21W",
    "product_locale": "us",
    "esci_label": "I",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Panasonic FV-20VQ3 WhisperCeiling 190 CFM Ceiling Mounted Fan",
    "product_description": null,
    "product_bullet_point": "WhisperCeiling fans feature a totally enclosed condenser motor and a double-tapered, dolphin-shaped bladed blower wheel to quietly move air\nDesigned to give you continuous, trouble-free operation for many years thanks in part to its high-quality components and permanently lubricated motors which wear at a slower pace\nDetachable adaptors, firmly secured duct ends, adjustable mounting brackets (up to 26-in), fan/motor units that detach easily from the housing and uncomplicated wiring all lend themselves to user-friendly installation\nThis Panasonic fan has a built-in damper to prevent backdraft, which helps to prevent outside air from coming through the fan\n0.35 amp",
    "product_brand": "Panasonic",
    "product_color": "White"
  },
  {
    "example_id": 1,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B07X3Y6B1V",
    "product_locale": "us",
    "esci_label": "E",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Homewerks 7141-80 Bathroom Fan Integrated LED Light Ceiling Mount Exhaust Ventilation, 1.1 Sones, 80 CFM",
    "product_description": null,
    "product_bullet_point": "OUTSTANDING PERFORMANCE: This Homewerk's bath fan ensures comfort in your home by quietly eliminating moisture and humidity in the bathroom. This exhaust fan is 1.1 sones at 80 CFM which means it\u2019s able to manage spaces up to 80 square feet and is very quiet..\nBATH FANS HELPS REMOVE HARSH ODOR: When cleaning the bathroom or toilet, harsh chemicals are used and they can leave an obnoxious odor behind. Homewerk\u2019s bathroom fans can help remove this odor with its powerful ventilation\nBUILD QUALITY: Designed to be corrosion resistant with its galvanized steel construction featuring a modern style round shape and has an 4000K Cool White Light LED Light. AC motor.\nEASY INSTALLATION: This exhaust bath fan is easy to install with its no-cut design and ceiling mount ventilation. Ceiling Opening (L) 7-1/2 in x Ceiling Opening (W) 7-1/4 x Ceiling Opening (H) 5-3/4 in. 13 in round grill and 4 in round duct connector.\nHOMEWERKS TRUSTED QUALITY: Be confident in the quality and construction of each and every one of our products. We ensure that all of our products are produced and certified to regional, national and international industry standards. We are proud of the products we sell, you will be too. 3 Year Limited",
    "product_brand": "Homewerks",
    "product_color": "80 CFM"
  },
  {
    "example_id": 2,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B07WDM7MQQ",
    "product_locale": "us",
    "esci_label": "E",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Homewerks 7140-80 Bathroom Fan Ceiling Mount Exhaust Ventilation, 1.5 Sones, 80 CFM, White",
    "product_description": null,
    "product_bullet_point": "OUTSTANDING PERFORMANCE: This Homewerk's bath fan ensures comfort in your home by quietly eliminating moisture and humidity in the bathroom. This exhaust fan is 1. 5 sone at 110 CFM which means it\u2019s able to manage spaces up to 110 square feet\nBATH FANS HELPS REMOVE HARSH ODOR: When cleaning the bathroom or toilet, harsh chemicals are used and they can leave an obnoxious odor behind. Homewerk\u2019s bathroom fans can help remove this odor with its powerful ventilation\nBUILD QUALITY: Designed to be corrosion resistant with its galvanized steel construction featuring a grille modern style.\nEASY INSTALLATION: This exhaust bath fan is easy to install with its no-cut design and ceiling mount ventilation. Ceiling Opening (L) 7-1/2 in x Ceiling Opening (W) 7-1/4 x Ceiling Opening (H) 5-3/4 in and a 4\" round duct connector.\nHOMEWERKS TRUSTED QUALITY: Be confident in the quality and construction of each and every one of our products. We ensure that all of our products are produced and certified to regional, national and international industry standards. We are proud of the products we sell, you will be too. 3 Year Limited",
    "product_brand": "Homewerks",
    "product_color": "White"
  },
  {
    "example_id": 3,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B07RH6Z8KW",
    "product_locale": "us",
    "esci_label": "E",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Delta Electronics RAD80L BreezRadiance 80 CFM Heater/Fan/Light Combo White (Renewed)",
    "product_description": "This pre-owned or refurbished product has been professionally inspected and tested to work and look like new. How a product becomes part of Amazon Renewed, your destination for pre-owned, refurbished products: A customer buys a new product and returns it or trades it in for a newer or different model. That product is inspected and tested to work and look like new by Amazon-qualified suppliers. Then, the product is sold as an Amazon Renewed product on Amazon. If not satisfied with the purchase, renewed products are eligible for replacement or refund under the Amazon Renewed Guarantee.",
    "product_bullet_point": "Quiet operation at 1.5 sones\nBuilt-in thermostat regulates temperature. Energy efficiency at 7.6 CFM/Watt\nPrecision engineered with DC brushless motor for extended reliability, this fan will outlast many household appliances\nGalvanized steel construction resists corrosion\nDuct: Detachable 4-inch Plastic Duct Adapter",
    "product_brand": "DELTA ELECTRONICS (AMERICAS) LTD.",
    "product_color": "White"
  },
  {
    "example_id": 4,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B07QJ7WYFQ",
    "product_locale": "us",
    "esci_label": "E",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Panasonic FV-08VRE2 Ventilation Fan with Recessed LED (Renewed)",
    "product_description": null,
    "product_bullet_point": "The design solution for Fan/light combinations\nEnergy Star rated architectural grade recessed Fan/LED light\nQuiet, energy efficient and powerful 80 CFM ventilation hidden above the Ceiling\nLED lamp is dimmable\nBeautiful Lighting with 6-1/2\u201daperture and advanced luminaire design",
    "product_brand": "Panasonic",
    "product_color": "White"
  },
  {
    "example_id": 5,
    "query": " revent 80 cfm",
    "query_id": 0,
    "product_id": "B076Q7V5WX",
    "product_locale": "us",
    "esci_label": "E",
    "small_version": 0,
    "large_version": 1,
    "split": "train",
    "product_title": "Panasonic FV-0511VQ1 WhisperCeiling DC Ventilation Fan, Speed Selector, SmartFlow Technology, Quiet,White",
    "product_description": null,
    "product_bullet_point": "Installation: Features a 4-inch or 6-inch duct adaptor ideal for new construction and renovations\nPrecision Spot Ventilation: Quiet and powerful ventilation while removing moisture and pollution\nPick-A-Flow Speed Selector: Allows you to pick desired airflow from 50, 80, 110 CFM\nFlexible Installation: Comes with Flex-Z Fast bracket for easy, fast and trouble-free installation\nEnergy Star Rated: Delivers powerful airflow without wasting energy",
    "product_brand": "Panasonic",
    "product_color": "White"
  }
]
```

File: README.md [hunk 1]
```diff
@@ -5,82 +5,85 @@
 Development environment for SolrCloud using Docker and Docker compose.
 This repository provides customized build for any lucene commit and solr commit.
 
-## How to use
+## How to use: create solr system
```
Comment (chlorochrule) [inline README.md]:
```suggestion
## How to use: Setup SolrCloud
```

File: README.md [hunk 2]
```diff
@@ -5,82 +5,85 @@
...
+2. generate dataset
+    ```bash
+    cd data
+    python3 create_shopping_data.py
+    cd ..
+    ```
+
+then you can check you json data in folder data/amazonshopping
+
+### launch dolr cloud
```
Comment (chlorochrule) [inline README.md]: typo
```suggestion
### launch SolrCloud
```

File: README.md [hunk 3]
```diff
@@ -5,82 +5,85 @@
...
+Run:
 
 ```bash
 docker-compose up -d --build
 ```
 
-and access `localhost:8983` or `localhost:8984`.
+You can see the solr system by accessing `localhost:8983` or `localhost:8984`.
+
+The pull down menu 「Collection Selector」is empty now.
```
Comment (chlorochrule) [inline README.md]: use ascii
```suggestion
The pull down menu "Collection Selector" is empty now.
```

File: README.md [hunk 4]
```diff
@@ -5,82 +5,85 @@
...
+The `field` defines every fields in json file, with the primary key named `example_id`.
 
-2. unzip dataset
-```bash
-du wikipedia_ja.zip
-# 2841736 wikipedia_ja.zip
-
-md5 wikipedia_ja.zip
-# MD5 (wikipedia_ja.zip) = f90b2dcf8e640fda7ac942c95136cd40
+Now, access to `localohst:8983`, you can find the `amazonshopping` included in the pull down menu 「Collection Selector」.
```
Comment (chlorochrule) [inline README.md]:
```suggestion
Now, access to `localohst:8983`, you can find the `amazonshopping` included in the pull down menu "Collection Selector".
```

File: README.md [hunk 5]
```diff
@@ -5,82 +5,85 @@
...
 
-unzip wikipedia_ja.zip
-```
+※ Here, you can find solr API [here](https://solr.apache.org/guide/7_1/coreadmin-api.html#coreadmin-create).
```
Comment (chlorochrule) [inline README.md]:
```suggestion
You can find Solr API [here](https://solr.apache.org/guide/solr/latest/configuration-guide/coreadmin-api.html#coreadmin-create)
```

File: README.md [hunk 6]
```diff
@@ -5,82 +5,85 @@
...
+### Feed documents
 
-3. Feed using curl
+The json file prepared before can be fee by curl command:
```
Comment (chlorochrule) [inline README.md]: typo
```suggestion
The json file prepared before can be feed by curl command:
```

File: data/create_shopping_data.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    df_products,
+    how='left',
+    left_on=['product_locale','product_id'],
+    right_on=['product_locale', 'product_id']
+)
+
+try:
+    os.mkdir(OUTPUT_FOLDER)
+except FileExistsError:
+    pass
```
Comment (chlorochrule) [inline data/create_shopping_data.py]: `os.makedirs` with `exist_ok=True` option can make it simple!
https://minus9d.hatenablog.com/entry/2014/12/26/232423

**Review by chlorochrule — COMMENTED** [review]
@ukohank517 Thank you for contribution! I left a bit of comment :)

File: data/create_shopping_data.py (hunk 7 above)
Comment (ukohank517) [reply data/create_shopping_data.py]: Oh, I never knew that before!
I'll use that function without try-catch block

**Comment by chlorochrule** [comment]
@ukohank517 LGTM, Thank you for contributing!

**Review by chlorochrule — APPROVED** [review]

**Pull request merged**